history/views.py

```diff
@@ -46,4 +54,33 @@ def get(self, *args, **kwargs):
         serializer = self.serializer_class(
             self._get(self._model, **kwargs))
 
-        return Response(serializer.data, status=200)
\ No newline at end of file
+        return Response(serializer.data, status=200)
+
+class StandupByWeek(Query, ListAPIView):
+    """ feed endpoint.
+        contains scheduled events, daily report, etc.
+    """
+    queryset = None
+    serializer_class = ShortStandupProjectSerializer
+    pagination_class = WeeklyReportsPagination
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        # get date parameter from url
+        dt = self.kwargs['date']
+        # convert date parameter to date value
+        current_date = datetime.strptime(dt, "%Y-%m-%d").date()
+        # compute the start of the week value
+        start_of_week = current_date - timedelta(days=current_date.weekday())
```

Comment:
create a method in `utils.mixins.TZ` class, maybe named it `dt_range(start, interval)` and put your date range code in there. we can reuse it everytime we need to calculate for date range.